### PR TITLE
Log 'panic' when using upstart

### PR DESCRIPTION
**Pull request by hush-hush**
### What does this PR do?

We don't have a way to alway log panics from the agent itself. So with
upstart we need to activate the 'log' option for the console. This way
panic logs will end up in /var/log/upstart/datadog-agent.log

By default systemd will redirect stderr to the journal, so we don't have
to specify anything.

### CODE REVIEW

**Comment by olivielpeau** [comment]
If I understand correctly this would make upstart log all of the agent's output to `/var/log/upstart/datadog-agent.log`, and upstart would take care of the log rotation (actually, `logrotate`, not upstart directly).

The one thing I don't like about this is that it doesn't seem to be possible to change the `logrotate` configuration that's used for the upstart logs (https://unix.stackexchange.com/questions/343842/how-to-override-logrotate-for-specific-upstart-service), so we would use `logrotate`'s default configuration, which takes into consideration the age of the log files and the maximum number of log files, not the actual size of the log files (which is what we do with `seelog` and in agent 5).

This behavior is not good IMO: the agent can log a lot of entries, so we really need to limit the actual size of the log files.

As an alternative to logging everything with upstart, could we only log the panics/crashes? Unless the agent is completely broken this would naturally limit the size of the upstart log files. I think we could do that by making `seelog` not output to stdout when the agent runs as a service and/or playing with redirections of stdout/stderr in the `exec` of the upstart job.

What do you think?

**Comment by hush-hush** [comment]
@olivielpeau I think you're right. I added an option to disable console logging (which might be useful in other cases). I did not document it in the `datadog.yaml` has I don't see any use cases when users would use it.

Let me know what you think.

File: omnibus/config/templates/datadog-agent/upstart.conf.erb [hunk 1]
```diff
@@ -5,12 +5,14 @@ stop on runlevel [!2345]
 
 respawn
 
-console none
+console log
 
 script
+  # Log to console is disable since the agent already logs using file or syslog depending on the configuration.
```
Comment (olivielpeau) [inline omnibus/config/templates/datadog-agent/upstart.conf.erb]: typo: `disable` -> `disabled`

**Review by olivielpeau — DISMISSED** [review]
looks good! 👌 

Just a small typo, feel free to merge once it's fixed :)

**Pull request merged**